extras/kms/kms.go

```diff
@@ -289,55 +289,169 @@ func (k *Kms) CreateKeys(ctx context.Context, scopeId string, purposes []KeyPurp
 		opts.withRandomReader = rand.Reader
 	}
 
+	r, w, localTx, err := k.txFromOpts(ctx, opt...)
+	if err != nil {
+		return fmt.Errorf("%s: %w", op, err)
+	}
+
+	if _, err := createKeysTx(ctx, r, w, rootWrapper, opts.withRandomReader, scopeId, purposes...); err != nil {
+		if localTx != nil {
+			if rollBackErr := localTx.Rollback(ctx); rollBackErr != nil {
+				err = multierror.Append(err, rollBackErr)
+			}
+		}
+		return fmt.Errorf("%s: %w", op, err)
+	}
+	if localTx != nil {
+		if err := localTx.Commit(ctx); err != nil {
+			return fmt.Errorf("%s: unable to commit transaction: %w", op, err)
+		}
+	}
+
+	return nil
+}
+
+// RotateKeys will rotate the root key versions and all DEKs for the current
+// KeyPurpose(s) of the KMS.
+//
+// If an optional WithRewrap(...) is request, then all keys will be re-encrypted
+// with the current root key wrapper.
+//
+// WithTx(...) and WithReaderWriter(...) options are supported which allow the
+// caller to pass an inflight transaction to be used for all database
+// operations.  If WithTx(...) or WithReaderWriter(...) are used, then the
+// caller is responsible for managing the transaction.  If neither WithTx or
+// WithReaderWriter are specified, then RotateKeys will rotated the scope's keys
+// within it's own transaction, which will be managed by RotateKeys.
+//
+// The WithRandomReader(...) option is supported.  If no optional random reader
+// is provided, then the reader from crypto/rand will be used as a default.
+//
+// Options supported: WithRandomReader, WithTx, WithRewrap
```

Comment:
Given the above comments I assume `WithReaderWriter` should be here too
```suggestion
// Options supported: WithRandomReader, WithTx, WithReaderWriter, WithRewrap
```

Reply:
ty!